Objects/dictobject.c

```diff
@@ -614,6 +614,60 @@ new_dict_with_shared_keys(PyDictKeysObject *keys)
     return new_dict(keys, values);
 }
 
+
+static PyObject *
+clone_dict(PyObject *s)
+{
+    PyDictKeysObject *keys;
+    PyDictObject *new;
+    PyDictObject *ms;
```

Comment:
I prefer `old`, `orig`, or `source` to `ms`.
And I prefer "declare and initialize at same time" style because we can use C99 syntax.